packages/compass-field-store/src/stores/store.ts

```diff
@@ -13,26 +14,30 @@ export function activatePlugin(
 ) {
   const store = createStore(reducer);
 
-  const emitFieldsChanged = (ns: string) => {
-    const fieldsState = store.getState()[ns];
+  const emitFieldsChanged = (connectionInfo: ConnectionInfo, ns: string) => {
+    const namespacesState = store.getState()[connectionInfo.id] ?? {};
+    const fieldsState = namespacesState[ns];
     globalAppRegistry.emit('fields-changed', {
+      connectionInfo,
       ns,
       ...fieldsState,
       autocompleteFields: schemaFieldsToAutocompleteItems(fieldsState.fields),
     });
   };
 
   const onDocumentsChanged = async ({
+    connectionInfo,
     ns,
     docs,
   }: {
+    connectionInfo: ConnectionInfo;
     ns: string;
     docs: Document[];
   }) => {
     try {
       const { fields } = await parseSchema(docs);
-      store.dispatch(changeFields(ns, fields));
-      emitFieldsChanged(ns);
+      store.dispatch(changeFields(connectionInfo.id, ns, fields));
```

Comment:
I was earlier checking for connectionInfo and throwing an error if it was not provided and later removed it but I am still a little worried that at some point someone could just emit an event on globalAppRegistry without passing the connectionInfo.

The defacto moving forward should be that if we ever want to emit an event from within the connected application, we should do that via ConnectionScopedGlobalAppRegistry. Does it make sense to have maybe an eslint plugin taking care of that? Or should it be more strict like throwing an error or something in globalAppRegistry if we come across an event that should instead be emitted via ConnectionScopedGlobalAppRegistry?

Reply:
As we discussed, moving forward we should not be using global or local app registry for these types of interaction, I'm not sure how much time we want to invest in polishing a mechanism that we should throw away eventually

Reply:
I would suggest to just log an error and throw in case the connectionInfo is not provided. Eventually we won't use events in the globalAppRegistry, but we would like to keep track of possible issues, and logging is probably the cheapest way that won't break anything.

Ideally if there is something missing we want to be aware of it, and we don't have a way to track errors in the background in production, so fast failing seems to be the more appropriate way.